contracts/pcv/PCVDepositAggregator.sol

```diff
@@ -0,0 +1,444 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import "./IPCVDepositAggregator.sol";
+import "./PCVDeposit.sol";
+import "../refs/CoreRef.sol";
+import "../external/Decimal.sol";
+import "./balancer/IRewardsAssetManager.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import "../libs/UintArrayOps.sol";
+
+/// @title PCV Deposit Aggregator
+/// @notice A smart contract that aggregates erc20-based PCV deposits and rebalances them according to set weights 
+contract PCVDepositAggregator is IPCVDepositAggregator, PCVDeposit {
+    using EnumerableSet for EnumerableSet.AddressSet;
+    using SafeERC20 for IERC20;
+    using SafeCast for uint256;
+    using SafeCast for int256;
+    using Decimal for Decimal.D256;
+    using UintArrayOps for uint256[];
+
+    // ---------- Properties ----------
+
+    EnumerableSet.AddressSet private pcvDepositAddresses;
+    mapping(address => uint256) public pcvDepositWeights;
+    
+    // Bufferweights is the weight of the aggregator itself
+    uint256 public bufferWeight;
+
+    // Totalweight is the sum of all deposit weights + the buffer weight
+    uint256 public totalWeight; 
+
+    // The token that this aggregator deals with. Cannot be changed.
+    address public immutable override token;
+
+    // The asset manager that controls the rewards for this aggregator
+    address public override assetManager;
+
+    constructor(
+        address _core,
+        address _assetManager,
+        address _token,
+        address[] memory _initialPCVDepositAddresses,
+        uint128[] memory _initialPCVDepositWeights,
+        uint128 _bufferWeight
+    ) CoreRef(_core) {
+        require(_initialPCVDepositAddresses.length == _initialPCVDepositWeights.length, "Addresses and weights are not the same length!");
+        require(_assetManager != address(0x0), "Rewards asset manager cannot be null");
+        require(IRewardsAssetManager(_assetManager).getToken() == _token, "Rewards asset manager must be for the same token as this.");
+
+        assetManager = _assetManager;
```

Comment:
assetManager should use an internal setter

Reply:
already talked about this - can't do this because you can't read immutable vars in the constructor